### ISSUE TITLE
Arabic words are not properly printing

### ISSUE THREAD

**Issue by GoogleCodeExporter** [python-escpos/python-escpos#37]
```
when we are printing the arabic words on Epson TM-T20 printer, it is not 
properly printing. Is it the problem with python module or the problem with 
printer.

Os: Raspberry Pi


```

Original issue reported on code.google.com by `dini...@gmail.com` on 20 Oct 2014 at 10:21

Attachments:
- [current.JPG](https://storage.googleapis.com/google-code-attachments/python-escpos/issue-37/comment-0/current.JPG)
- [expected.JPG](https://storage.googleapis.com/google-code-attachments/python-escpos/issue-37/comment-0/expected.JPG)

**Comment by Davidamgad2** [comment]
Greetings,

Does anyone figure out to print Arabic words? I have been search for a while now, and I couldn't find anything.
Thanks in advance! 

Regards.

**Comment by belono** [comment]
Hi @Davidamgad2 .

Are you setting the charcode to your printer?
There is a command that changes the charcode page of your printer, but your printer must support that charcode.
Hope this helps: [#444](https://github.com/python-escpos/python-escpos/issues/444#issuecomment-767147027)

We are glad to help but, please, provide more info to let us help you better.
We need to know your system environment (SO, python version, python-escpos version, ...), your printer model and a snippet that reproduces the issue.

Thank you ;)

**Comment by Davidamgad2** [comment]
Hello @belono,

Thanks for your response! I appreciate your time.

I read the #444 and followed the solution there. 

I have search for the codepage for Arabic and I found that it's CP720 from [here](https://en.wikipedia.org/wiki/Code_page_720)

Weird thing that my printer I found the code page for from [here ](https://united-egypt.net/products/xp-d200n)

It says that it's

The issue here that when I try to print my sentence it doesn't give it to me. 

This is the code I try 
`# -*- coding: utf-8 -*-

from escpos.printer import Usb

p = Usb(0x1FC9, 0x2016)
p.charcode('CP720')
p.text( "ازيك؟\n")
p.barcode('1324354657687', 'EAN13', 64, 2, '', '')
p.cut()
p.close()`

Device info:
OS: Windows 10 Pro
Python: 3.10.6
python-escpos version: I'm not sure but I follow the instruction that you gave for the prelease version(I think it's python_escpos-3.0a8)

My printer is 

Xprinter XP-D200N

Thank you so much for your time helping me with this. I also apologize if there's any random info. I'm new to the thermal printer. :")

![snippts](https://user-images.githubusercontent.com/111524016/210327541-554011ac-c628-4a69-8971-8f68414203ed.jpg)

**Comment by Davidamgad2** [comment]
I had to figure out a solution and used Arabic shaper and and Image method to solve this issue.

**Comment by belono** [comment]
Hi @Davidamgad2 .

When trying the following commands, I'm getting this output:
```
>>> from escpos.printer import Dummy
>>> dummy = printer.Dummy(profile="default")
>>> dummy.text("ازيك؟\n")
>>> print(dummy.output.decode('CP720'))
ازيك?
```
Is this output the one that is expected?
If so, try to add the param `profile="default"` to the Usb connector, then 
```
from escpos.printer import Usb
p = Usb(0x1FC9, 0x2016, profile="default")
p.text("ازيك؟\n")
p.close()
```
Are the correct characters printed now?

**Comment by Davidamgad2** [comment]
Hello @belono 

Thanks for your reply! 

I have tried your solution but the same happened it didn't give me the characters I want. It gave me the one in pic I previously uploaded.

Further more I tried  
`>>> from escpos.printer import Dummy
>>> dummy = printer.Dummy(profile="default")
>>> dummy.text("ازيك؟\n")
>>> print(dummy.output.decode('CP720'))
ازيك?`
It didn't give me the same output as it gave you. Have you installed any further packages?

Regards.

**Comment by belono** [comment]
> It didn't give me the same output as it gave you. Have you installed any further packages?


No, I haven't. I'm testing on a fresh installation in a pyenv virtualenv.

What about this:
```
from escpos.printer import Dummy
dummy = Dummy()
dummy.charcode("CP720")
dummy.text("ازيك؟\n")
print(dummy.output.decode("CP720"))
```
Are you getting the right output now? If so, try sending the dummy output to your printer:
```
p = Usb(0x1FC9, 0x2016)
p._raw(dummy.output)
p.close()
```
If the first worked, but didn`t the second, the problem may be in your printer's codepage.
What happens if you set the profile now?
```
p = Usb(0x1FC9, 0x2016, profile="default")
p._raw(dummy.output)
p.close()
```
